DMX.Sdk.Tests.Unit/Services/Foundations/Labs/LabServiceTests.Exceptions.RetrieveAll.cs

```diff
@@ -0,0 +1,144 @@
+﻿// --------------------------------------------------------------- 
+// Copyright (c) Microsoft Corporation. All rights reserved. 
+// ---------------------------------------------------------------
+
+using DMX.Sdk.Models.Services.Foundation.Exceptions;
+using DMX.Sdk.Models.Services.Foundation.Labs.Exceptions;
+using DMX.Sdk.Models.Services.Foundations.Labs;
+using FluentAssertions;
+using Moq;
+using RESTFulSense.Exceptions;
+using Xeptions;
+using Xunit;
+
+namespace DMX.Sdk.Tests.Unit.Services.Foundations.Labs
+{
+    public partial class LabServiceTests
+    {
+        [Theory]
+        [MemberData (nameof(CriticalDependencyException))]
+        public async Task ShouldThrowCriticalDependencyExceptionOnRetrieveAllIfCriticalErrorOccursLogItAsync(
+            Xeption criticalDependencyException)
+        {
+            // given
+            var failedLabDependencyException =
+                new FailedLabDependencyException(criticalDependencyException);
+
+            var expectedLabDependencyException =
+                new LabDependencyException(failedLabDependencyException);
+
+            this.dmxApiBroker
+                .Setup(broker => broker.GetAllLabsAsync())
+                .ThrowsAsync(criticalDependencyException);
+
+            // when
+            ValueTask<List<Lab>> getAllLabsTask =

```

Comment:
Variable name should match method name.
This variable should be called `retrieveAllLabsTask` 

Do the same for the rest.